internal/commands/push/command_test.go

```diff
@@ -1029,11 +1029,135 @@ Try instead: realm-cli push --local testdata/dependencies --remote appID --inclu
+		appData        local.AppData
+		expectedConfig string
+	}{
+		{
+			appConfig: local.FileRealmConfig,
+			appData: &local.AppRealmConfigJSON{local.AppDataV2{local.AppStructureV2{
+				ConfigVersion:   realm.AppConfigVersion20210101,
+				Name:            "eggcorn",
+				Location:        realm.Location("location"),
+				DeploymentModel: realm.DeploymentModel("deployment_model"),
+				Environment:     realm.Environment("environment"),
+			}}},
+			expectedConfig: `{
+    "config_version": 20210101,
+    "app_id": "eggcorn-abcde",
+    "name": "eggcorn",
+    "location": "location",
+    "deployment_model": "deployment_model",
+    "environment": "environment"
+}
+`,
+		},
+		{
+			appConfig: local.FileConfig,
+			appData: &local.AppConfigJSON{local.AppDataV1{local.AppStructureV1{
+				ConfigVersion:   realm.AppConfigVersion20200603,
+				Name:            "eggcorn",
+				Location:        realm.Location("location"),
+				DeploymentModel: realm.DeploymentModel("deployment_model"),
+				Environment:     realm.Environment("environment"),
+			}}},
+			expectedConfig: `{
+    "config_version": 20200603,
+    "app_id": "eggcorn-abcde",
+    "name": "eggcorn",
+    "location": "location",
+    "deployment_model": "deployment_model",
+    "environment": "environment",
+    "security": null,
+    "custom_user_data_config": {
+        "enabled": false
+    },
+    "sync": {
+        "development_mode_enabled": false
+    }
+}
+`,
+		},
+		{
+			appConfig: local.FileStitch,
+			appData: &local.AppConfigJSON{local.AppDataV1{local.AppStructureV1{
+				ConfigVersion:   realm.AppConfigVersion20180301,
+				Name:            "eggcorn",
+				Location:        realm.Location("location"),
+				DeploymentModel: realm.DeploymentModel("deployment_model"),
+				Environment:     realm.Environment("environment"),
+			}}},
+			expectedConfig: `{
+    "config_version": 20180301,
+    "app_id": "eggcorn-abcde",
+    "name": "eggcorn",
+    "location": "location",
+    "deployment_model": "deployment_model",
+    "environment": "environment",
+    "security": null,
+    "custom_user_data_config": {
+        "enabled": false
+    },
+    "sync": {
+        "development_mode_enabled": false
+    }
+}
+`,
+		},
+	} {
+		t.Run(fmt.Sprintf("with a %s config file", tc.appConfig.String()), func(t *testing.T) {
+			tmpDir, teardown, tmpDirErr := u.NewTempDir("push_handler")
+			assert.Nil(t, tmpDirErr)
+			defer teardown()
+
+			app := local.App{RootDir: tmpDir, Config: tc.appConfig, AppData: tc.appData}
+
+			app.WriteConfig()
+
+			out := new(bytes.Buffer)
+			console, ui, consoleErr := mock.NewConsoleWithOptions(mock.UIOptions{AutoConfirm: true}, out)
+			assert.Nil(t, consoleErr)
+			defer console.Close()
+
+			cmd := &Command{inputs{LocalPath: filepath.Join(tmpDir, "nested"), RemoteApp: "appID"}}
+			err := cmd.Handler(nil, ui, cli.Clients{Realm: realmClient})
+
+			assert.Nil(t, err)
+			configData, readErr := ioutil.ReadFile(filepath.Join(tmpDir, tc.appConfig.String()))
```

Comment:
[nit/opt] I know this check tells us that the config file was written in the app's root directory but it might also be worth checking that no config file was written to the nested dir 

before this fix, it was previously possible to have an existing config in the root dir and then on `push --local <nested>`, an additional config would be written to the nested dir

Reply:
Oh yeah, I forgot that was happening. I'll add that!